### ISSUE TITLE
[PNI] Reduce queries for buyer's guide page without products

### ISSUE THREAD

**Issue by tbrlpld** [MozillaFoundation/foundation.mozilla.org#9814]
## Description

This is a follow-up ticket to: #9545. 

While not strictly dependent on them, this ticket should be tackled after #9812 and #9813, because the other two should allow for much greater performance improvements. 

The performance investigation in #9545 showed that it requires 156 database queries to render the buyer's guide homepage without even a single product being displayed. 

Without any products, this page has practically no content. 

This ticket should further investigate and improve (reduce) the number of database queries to render this practically empy page.

Since the cause of the large number of queries is not obvious, this ticket might need to be a spike.

The use of the Django Debug Toolbar should be helpful in further investigating what generates all these queries. I have been experiencing issues with the DjDT, because it's state seems to be wiped by our buyer's guide JS scripts. For the investigation it might be helpful to deactivate the buyer's guide JS. 

## Acceptance criteria

(The below query targets are only rough goals for what should be more "normal performance", but really the lower the better. Keep in mind though, the principle of [diminishing returns](https://en.wikipedia.org/wiki/Diminishing_returns). At some point the effort won't justify saving 1 or 2 more queries. But it seems like we still have some way to go before we get into that territory.)

- [ ] Soft target: As a developer, I can confirm from the unittests that it takes less than 100 queries to render the buyer's 
guide homepage without any product on it.
- [ ] Harder target: As a developer, I can confirm from the unittests that it takes less than 50 queries to render the buyer's guide homepage without any product on it.

**Comment by mmmavis** [comment]
Done! 🎉
